### DISCUSSION TITLE
스터디 2기 첫 번째 회고

### DISCUSSION THREAD

**Discussion by DaleSeo** [DaleStudy/leetcode-study#421] [건의사항]
스터디 2기 참여자 여러분, 안녕하세요.

건강한 스터디 문화를 조성하고 멤버분들과 코치님들께 좋은 경험을 드리기 위해 5주에 한 번씩 회고를 하여 스터디 운영에 반영을 하고 있습니다. 모임 때 말씀드렸던 것처럼 온라인 모임 때 문제 설명 때문에 시간이 부족할 것 같아서 비동기로 피드백을 취합하려고 합니다. 첫 4~5주간의 스터디 경험에 대해서 본 디스커션 글에 댓글로 달아주시고, 공감이 되는 다른 댓글에 Upvote를 해주세요. 대댓글로 추가 소통을 해도 좋을 것 같네요. 디스커션에서 Upvote를 많이 받거나 대댓글이 많이 달린 피드백을 위주로 모임 때 함께 얘기해보면 좋을 것 같습니다.

## 댓글 형식
- 좋았던 점과 아쉬웠던 점을 구분하기 위해서 [Went Well] 또는 [To Improve] 말머리를 달아주세요. (예: [Went Well] 단순히 문제만 푸는 것이 아니라 다른 코드를 리뷰하고 답안 설명을 하면서 실력이 늘었습니다.)
- 아쉬웠던 점에 대해서 피드백을 주실 때는 함께 실천해볼 수 잇는 개선 방안도 포함시켜주시면 큰 도움이 될 것 같습니다.
- https://github.com/DaleStudy/leetcode-study/discussions/211 에서 스터디 1기의 회고 기록도 볼 수 있으니 참고 바랍니다.

**Comment by DaleSeo** [comment]
[Went Well] 아직 초반이지만 1기 대비 포기하신 분들이 많지 않아서 너무 좋습니다! 생존자 20명 이상으로 쭈욱 유지되길 바래봅니다. 🙏

**Comment by DaleSeo** [comment]
[To Improve] 디스코드에 음성 채널에 인원 제한이 있는지 몰라서 첫 모임 때 온보딩이 매끄럽지 못해서 정말 죄송했습니다.
- 개선안: 스테이지 채널 기능을 이용하면 무료로 최대 50명까지 입장이 가능하다니, 3차 첫 모임 때는 활용하려고 합니다.

**Comment by Sunjae95** [comment]
[Went Well] 혼자라면 매주 5개씩 풀지 못했을텐데 스터디라는 동기부여..? 4주차지만 덕분에 문제를 20개가량 풀어서 많이 만족하고 있습니다.

**Comment by Sunjae95** [comment]
[To Improve] 문제풀이 완료시점처럼 리뷰완료시점이 있으면 좋겠습니다.
1주일동안 다들 열심히 푸시고 정기모임에서 진행상황을 볼 때 merge가 되어있으면 더 뿌듯하고 스터디가 잘 진행된다는 느낄 수 있을거 같아요. 
이 부분이 스터디에서 부담스러운 룰이 될 수 있어서 정기모임전 12시간으로 부담을 최소화하여 조심히 피드백 및 건의올립니다 :)

- 개선안: 스터디 시작 12시간전 / 한국시간기준 매주 토요일 저녁 10시까지 (해외분들 죄송합니다..!)

**Reply by DaleSeo** [reply]
@Sunjae95 님, 저도 비슷한 생각을 하고 있었는데 의견을 주셔서 감사합니다.
코드 리뷰가 토요일 저녁 10시까지 끝나는 게 잘 지켜지려면, 그 전에 답안 제출도 기한도 필요하지 않을까요? (예를 들어, 금요일 저녁 10시)

**Reply by DaleSeo** [reply]
@DaleSeo 설문 조사

**Comment by TonyKim9401** [comment]
[Went Well] 평소에 혼자 알고리즘 문제를 풀다가 같이 푸는 동료가 생기고 또한 리뷰를 해주실 분이 생기셨다는거에 의욕을 얻고 있습니다. 그리고 가능한 매주 발표를 하려고 하고있습니다. 발표를 통해 `대충 이렇게 풀면 되겟지?` 라고 생각했던 수준을 넘어, 다른 사람에게 해결한 문제의 본질을 전달하고 가능한 쉽게 전달하려고 풀어나가는 것 또한 스스로 많은 공부가 되고있어 더 깊이 각인되고 있다고 생각합니다.

**Reply by DaleSeo** [reply]
@TonyKim9401 님, 모임 시간에 자신의 답안 설명함으로써 제가 의도했던 이점을 너무 잘 얻어가시는 것 같아서 흐뭇하네요! 😄

**Comment by DaleSeo** [comment]
[Went Well] 파일 마지막에 개행이 없는 파일에 대해서 코드 리뷰할 때 더 이상 신경쓰지 않아도 되서 좋네요. 또한 PR에 언어 레이블이 자동으로 달려서 편해진 것 같아요. GitHub Actions 작업을 추가해주신 @bky373 코치 님의 헌신에 감사드립니다! 🙇

**Reply by bky373** [reply]
처음에 우여곡절이 있었지만 안정화를 위해 힘써주셔서 감사합니다 @DaleSeo 님!
그리고 @TonyKim9401 님의 제안과 @sounmind 님의 제안이 큰 도움이 되었습니다 두 분께도 감사드립니다! 🙇

**Comment by highballplz** [comment]
[Went Well] 

디스커션에 올린 질문들에 다른 스터디원 분들의 소중한 답변을 받을 수 있어 좋았습니다.

**Comment by highballplz** [comment]
[To be improved] 

개선점이라기보다는 희망사항인데, 이미 해외 현직에 계신 분들의 AMA(Ask Me Anything)를 열어주시면 정말 감사하겠습니다. 

물론 스터디 끝나고 자유롭게 얘기할 시간이 있기는 하지만, 한 분이 마이크를 잡으시고 현지 생활 얘기를 들어볼 수 있는 기회가 된다면 저로서는 큰 동기 부여가 될 것 같습니다.

**Reply by yolophg** [reply]
말씀하신대로 이런 세션을 만들어 기여할 수 있는 별도 공간이 있으면 좋을 듯 합니다 :)

**Comment by kim-young** [comment]
[Went Well] 다 같이 함에서 오는 동기 부여는 물론이거니와, 문제를 풀고 나서 스스로 검토할수 있는 달레님의 풀이 링크가 제공되어 있는 부분이 저는 상당히 편하고 Accessible 했던거 같습니다. Leetcode에서 문제 풀이 후 Solution/Discussion 탭에 들어가도 가독성이 떨어질 뿐만 아니라 가끔 신빙성이 떨어지는 부분이 없지 않아 있었는데 (물론 훌륭한 정답들도 많았습니다!) 달레님께서 상세하게 풀어서 설명해주셔서 많이 배우는 것 같습니다. 특히 DP...!

**Comment by kim-young** [comment]
[To Be Improved] 얼마전 코드 리뷰 배정 받았을때 느낀 부분인데, 제가 잘 모르는 언어일 경우에는 조금 난감하더라고요 😭 물론 코드 보면 Context는 이해가 가고 그러면서 또 많이 배웠지만, 코드 리뷰 받는 분을 제가 좀 더 세밀하게 도와드리지 못하는 것 같아서 죄송했습니다! 다행히 코치분들께서 리뷰를 도와주셔서 잘 넘어갔지만, 한번쯤 상의 해볼만한 내용이 아닌가 싶습니다!

**Reply by jaejeong1** [reply]
저도 동감합니다. 제가 경험해보지 못한 언어일 경우 언어 특성이 포함된 코드 리뷰는 어려운 점이 있었습니다..

**Reply by DaleSeo** [reply]
어떤 개선방안이 있을까요? 코드 리뷰 책임제를 손 봐야할까요? 
저는 개인적으로 코드 리뷰의 깊이를 떠나서 여러분이 서로의 코드를 리뷰 해주신 것만 해도 감사하고 있습니다. 1기 때는 기본적인 PR 승인도 잘 안 되서 저의 코드 리뷰 부담이 상당했던 걸로 기억합니다 ㅠㅠ 2기는 초반이라서 단정짓기는 아직 이르지만, 뒤로 갈 수록 낙오하시는 분들이 늘어나기 때문에 지금처럼 멤버 간에 상호 코드 리뷰를 끝까지 유지만 되도 저는 매우 만족할 것 같습니다 ㅎㅎ

**Reply by obzva** [reply]
언어 특성이 덜 드러나게끔 코드를 작성하도록 노력하는게 좋을까요..?
(해당 언어의 고유한 syntax를 덜 사용하도록 노력한다든지)

**Reply by wogha95** [reply]
저는 다른 언어 풀이를 보면서 처음에 난감했었지만 새로운 언어의 문법이나 새로운 부분들을 알아갈 수 있어서 좋았습니다!
그래도 익숙한 언어보다는 이해가 느리긴했어서 로직 설명을 주석으로만 남겨도 이해에 엄청 도움될 듯 합니다!

**Reply by obzva** [reply]
언어가 다르더라도 문법의 큰 틀은 대부분 비슷하니까 주석 정도만 남겨도 큰 도움이 되겠네요 :D

**Reply by DaleSeo** [reply]
@DaleSeo 답안 제출 가이드에 관련 내용 추가

**Comment by obzva** [comment]
[Went Well]
발표 준비를 하고, 다른 분들의 발표에 의견을 제시하기 위해서 혼자 공부할 때보다 더 많은 공부를 해야 했습니다
덕분에 약한 부분을 찾고 보완할 수 있는 기회가 되는 것 같습니다

**Comment by obzva** [comment]
[To Be Improved]
DP 풀이를 발표할 땐 점화식에 대한 언급을 해주시는게 좋을 것 같습니다
https://discord.com/channels/775115965964222492/1229860988170403901/1279628653566886000

**Comment by DaleSeo** [comment]
[To Improve] 화상 모임 때 카메라를 꺼두시는 분들이 많아서 뵌지 몇 주가 지났는데 아직 얼굴도 모르는 분들이 있어서 개인적으로 아쉬웠습니다. 멤버 간에 친밀감 형성을 위해서 가급적 우리 카메라를 키고 대화하면 어떨까요? 의사소통에 있어서 사실 말이 차지하는 비중은 매우 적고 표정, 눈빛, 제스쳐 같은 비언어적 요소가 높다고 들었거든요. 비대면 커뮤니케이션의 질을 향상하는데도 분명히 도움이 되는 부분이 있을거라 생각합니다.

**Reply by obzva** [reply]
동의합니다! 저부터 켜겠습니다 ㅎㅎ

**Comment by jaejeong1** [comment]
[To Improve] 문제풀이 시 효과적인 리뷰를 위해 풀이 도출 과정이 좀 더 상세히 표기되었으면 합니다. 
상세히 라는 기준은 개인마다 다를 수 있으므로, 좋은 예시 템플릿을 공유해 이 템플릿을 따르도록 한다면 좋을 것 같습니다.

**Reply by obzva** [reply]
재정님께서 생각하시는 좋은 예시를 한 번 보여주시면 같이 이야기 나눠보기 더 좋을 것 같아용

**Reply by jaejeong1** [reply]
네, 좋습니다! 바로 찾아서 공유해보겠습니다 :)

**Reply by obzva** [reply]
저는 코멘트로 알고리즘의 전개 과정을 먼저 설명 드리고, 코드는 최대한 간결명료하게 적으려고 하고 있어요
https://github.com/DaleStudy/leetcode-study/pull/429

**Reply by jaejeong1** [reply]
# 알고리즘 문제 풀이 템플릿

## 1. 문제 요약
문제의 핵심 요구사항을 간략하게 요약합니다. 입력과 출력, 그리고 해결해야 할 목표를 명확하게 기술합니다.

## 2. 접근 방법
문제를 해결하기 위해 선택한 알고리즘적 접근을 설명합니다. 문제를 해결하기 위한 주요 아이디어와 알고리즘의 전반적인 흐름을 기술합니다.

- 문제를 해결하기 위한 단계:
  1. 문제를 해결하는 첫 번째 단계.
  2. 문제 해결을 위한 두 번째 단계.
  3. 필요한 추가 단계.

## 3. 알고리즘 과정

### 3.1 첫 번째 구간: 주요 연산 설명 (시간 복잡도 분석 포함)
문제 해결을 위한 첫 번째 구간을 설명합니다. 해당 구간에서 어떤 연산이 이루어지며, 이 구간의 시간 복잡도가 어떻게 도출되었는지 설명합니다.

- **시간 복잡도 설명:**
  - 이 구간에서 시간 복잡도는 O(X)로 나타납니다. 이를 도출한 이유를 설명합니다.

### 3.2 두 번째 구간: 주요 연산 설명 (시간 복잡도 분석 포함)
문제 해결을 위한 두 번째 구간을 설명합니다. 이 구간에서 이루어지는 연산과 복잡도 분석을 포함시킵니다.

- **시간 복잡도 설명:**
  - 이 구간에서 시간 복잡도는 O(Y)로 나타납니다. 복잡도 계산 과정을 설명합니다.

### 3.3 결과 반환
최종적으로 계산된 결과를 반환하는 부분을 설명합니다. 결과가 어떻게 도출되었는지 설명하고, 필요한 경우 시간 및 공간 복잡도를 추가로 설명합니다.

## 4. 시간 복잡도 분석
전체 알고리즘의 시간 복잡도를 설명합니다. 구간별로 달라지는 복잡도를 결합해 최종적인 복잡도를 도출합니다.

- 첫 번째 구간: O(X)
- 두 번째 구간: O(Y)
- 전체 시간 복잡도는 O(X + Y) 또는 O(max(X, Y)) 등으로 계산됩니다.

## 5. 공간 복잡도 분석
전체 알고리즘의 공간 복잡도를 설명합니다. 추가적으로 사용한 메모리 공간과 자료구조를 고려하여 복잡도를 도출합니다.

- 공간 복잡도는 O(Z)로 나타납니다. 추가 메모리 사용이 최소화되었는지 여부도 설명합니다.

## 6. 결론
문제의 요구사항을 충족하는지 간단히 요약하고, 추가적으로 고려해야 할 사항이나 최적화가 가능한 부분이 있는지 설명합니다.

**Reply by jaejeong1** [reply]
위 템플릿으로 작성한 예시

```
public class ComplexAlgorithm {
    /**
     * 1. 문제 요약
     * 예를 들어, 주어진 배열에서 특정 조건을 만족하는 서브배열을 찾는 문제입니다.
     * 서브배열의 길이와 값을 기반으로 최적의 값을 찾아야 하며, 다단계의 계산이 필요합니다.
     */

    /**
     * 2. 접근 방법
     * - 다단계로 문제를 나누어 해결:
     *   1. 주어진 배열에서 각 서브배열의 가능한 값을 계산.
     *   2. 각 서브배열에서 특정 조건을 만족하는 값만을 추출.
     *   3. 조건에 맞는 값들 중에서 최적의 값을 선택.
     */

    public int solveComplexProblem(int[] nums) {
        // 3. 알고리즘 과정
        // 3.1. 첫 번째 구간: 서브배열 생성 및 계산 (O(N^2))
        // 배열의 모든 가능한 서브배열을 계산하며, 각 서브배열의 합계를 구한다.
        int n = nums.length;
        int maxSum = Integer.MIN_VALUE;
        
        // 서브배열의 합을 구하는 이중 반복문 -> 시간 복잡도: O(N^2)
        for (int i = 0; i < n; i++) {
            int currentSum = 0;
            for (int j = i; j < n; j++) {
                currentSum += nums[j];
                // 각 서브배열에서 조건을 만족하는 최대 합을 추적
                if (currentSum > maxSum) {
                    maxSum = currentSum;
                }
            }
        }
        
        // 시간 복잡도 설명:
        // - 이중 루프가 N^2의 복잡도를 가집니다. 첫 번째 반복문은 N번 실행되고, 두 번째 반복문은 N번까지 실행되므로 O(N^2).

        // 3.2. 두 번째 구간: 특정 조건을 만족하는 값 필터링 (O(N))
        // 두 번째 구간에서는 계산된 결과를 기반으로 특정 조건을 만족하는 값을 추출합니다.
        // 예를 들어, 계산된 값이 양수일 때만 결과에 반영.
        int filteredSum = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] > 0) {
                filteredSum += nums[i];
            }
        }

        // 시간 복잡도 설명:
        // - 이 구간은 배열을 한 번 순회하므로 시간 복잡도는 O(N).
        
        // 3.3. 결과 반환
        // 최종적으로 계산된 값을 반환 (여기서는 maxSum을 반환).
        return Math.max(maxSum, filteredSum);
    }

    /**
     * 4. 시간 복잡도 분석
     * - 첫 번째 구간(서브배열 계산): O(N^2)
     * - 두 번째 구간(조건 필터링): O(N)
     * - 전체 시간 복잡도는 O(N^2) + O(N) = O(N^2)입니다.
     *   가장 높은 복잡도를 가진 구간이 전체 복잡도를 결정합니다.
     */

    /**
     * 5. 공간 복잡도 분석
     * - 추가적인 메모리 사용 없이 기존 배열을 기반으로 계산하므로 공간 복잡도는 O(1)입니다.
     * - 단, 만약 서브배열의 결과를 따로 저장해야 한다면 공간 복잡도가 O(N^2)로 증가할 수 있습니다.
     */
}
```

**Reply by jaejeong1** [reply]
@obzva 
GPT의 도움을 받아 퀵하게 초안 만들어보았습니다 :)
다같이 논의해보면서 제외/추가할 항목들 정해보면 좋을 것 같습니다!

**Reply by obzva** [reply]
오우 멋집니다 :) !!
퀵하게 제 개인 의견 남기겠습니다
1. 문제 요약
- 생략해도 될 것 같습니다 문제의 요구 사항은 문제 자체에 모두 기술되어 있기 때문입니다
- `이 문제는 사실 이걸 알아내는게 핵심이다` <- 이런 식의 내용을 설명하고자 한다면 `2` 접근 방법에 포함되면 좋을 것 같습니다
3. 코멘트는 필요하다면 적절히 사용하는 것 좋아보입니다 :D
4. (5.) Big-O 분석
- 시공간 분석은 한 챕터에서 하면 좋을 것 같습니다
- Big-O notation에 쓰이는 변수가 시공간 분석에 함께 쓰이는 경우가 많고, 둘의 설명이 연관되어 있는 경우가 많기 때문입니다

**Comment by SamTheKorean** [comment]
[To improve] 여러모로 바쁘긴 했지만 상황을 감안하더라도 코드리뷰나 알고리즘 풀이에 쓰는 시간이 많이 부족해서 건설적인 토론에 참여하지 못해서 아쉬웠습니다.

**Reply by yolophg** [reply]
동감합니다!

**Comment by SamTheKorean** [comment]
[What went well] 5 문제 풀이를 강요하지 않고 자율적으로 문제 풀이 하는게 지속성 측면에서 큰 장점인 것 같습니다!

**Comment by SamTheKorean** [comment]
[What went well] 처음부터 코드리뷰 정책을 도입해서 인지 코드리뷰 수준이 1기때보다도 전체적으로 상향 평준화 된 것 같습니다!

**Comment by SamTheKorean** [comment]
[What went well] 문제 설명 및 논의 하는 걸 듣고 보면서 다른 스터디원분들의 사고과정을 관찰하는게 도움이 많이 됐습니다.

**Comment by bky373** [comment]
[Went Well] 1기 때 저의 풀이와 2기 참여자 분들의 풀이를 비교해보는 과정이 재밌었고, 배울 게 많았습니다. 풀이에서 공통점이 있을 때는 제 풀이에 대한 확신을 가질 수 있었고, 차이가 있을 때는 최적화를 위한 힌트를 새롭게 얻을 수 있었습니다. 동일한 문제를 볼 때 앞으로 기억에 더 잘 남을 것 같습니다. (2기 분들에게 3기 코치 지원을 미리 적극 추천합니다 ㅎㅎ)

**Comment by yolophg** [comment]
[Went Well] 우리 스터디의 Discussion 공간을 다들 활발하게 이용해주셔서 더 적극적인 논의가 이뤄지는 것이 너무 좋았습니다.

**Comment by sounmind** [comment]
[Went Well] 1기 대비 참여자와 코드 리뷰가 활발하고, 정기 모임도 알고리즘 풀이와 논의로 활성화된 것 같아서 좋았어요. 긍정적인 자극이 됩니다!

**Comment by yolophg** [comment]
[Went Well & To improve] 문제 풀이 시간에 자율 참여임에도 불구하고, 많은 분들이 고퀄리티로 ppt까지 준비하셔서 진행되고 있어서 너무 감사하고 1기보다 더욱 발전되어서 굉장히 좋은 자극을 받을 수 있었습니다!
하지만 동시에 문제 풀이를 준비하는데에 부담을 가지셔서 쉽사리 지원하지 못하시는건 아닐까 + 매주 준비 시간에 많은 에너지를 쓰는게 아닐까 하는 생각도 들었습니다! 
따라서, 자율적으로 하시되 결국 궁극적 목표는 다시 한 번 내가 푼 문제를 이해하고 설명하는 것이니, 혹시 조금이라도 부담을 가지고 계시다면 좀 부담을 덜어내고 남은 스터디를 진행해도 좋을 것 같다는 말씀을 드리고 싶었습니다. 
(ex. 단순하게 코드 화면만 공유하고 주석처리로 설명하는걸로도 충분하다고 생각합니다)

**Reply by DaleSeo** [reply]
저도 사실 비슷한 생각을 하고 있었는데, 혹여 열심히 준비를 하시는 분들의 사기를 저하시킬까봐 망설이고 있었습니다.

처음에 모임 시간에 답안 설명을 계획한 본래 취지는 면접관에게 자신의 알고리즘을 설명하실 수 있는 연습 기회를 드리는 것이 었습니다. 실제 면접 환경에서 ppt와 같은 시각 자료를 사전에 준비갈 수 없고, 온라인 코드 에디터나 오프라인 면접이라면 화이트 보드 정도를 사용하실 수 있을 것입니다. 따라서 이렇게 제한된 환경에서 효과적으로 전달하는 설명하는 연습을 하는 게 더 실질적인 도움이 되실 것 같습니다. 너무 형식에 치중하거나 기대치를 올리는 것은 바람직하지 않으며, 서로에게 불필요한 부담이 될 수 있다고 생각합니다.

**Reply by obzva** [reply]
이렇게는 생각 못해봤는데, 공감합니다..

**Reply by DaleSeo** [reply]
@DaleSeo 정기 모임 문서에 Quality bar 명확하게 적기

**Comment by sounmind** [comment]
[To improve] 코치진들에게 일종의 보상이나 동기부여가 될 수 있는 게 있으면 코치님들의 리뷰도 더 활발해질 수 있을 것 같습니다. "이번 주에 좋았던 리뷰" 같은 것을 공유한다면 더 좋은 리뷰를 하기 위해 자연스럽게 노력하지 않을까요? 어쩌면 리뷰할 때 더 많은 부담이 될 수도 있겠네요.

**Reply by DaleSeo** [reply]
우리 지금 문제 설명할 분 취합하는 것처럼 다음 주 모임 공지 디스커션에 좋았던 리뷰를 댓글로 달아놓고 모임 시간에 다 같이 보고 칭찬하면 어떨까요?

**Reply by DaleSeo** [reply]
@SamTheKorean 다음 모임 공지부터 PR 피드백 링크 공유해달라는 내용 추가해주세요.

**Comment by obzva** [comment]
[Went Well]
시간 나면 제 담당이 아닌 분들의 풀이도 쭉 한 번 보는데 이것 덕분에 평소에 볼 일 없는 언어도 구경하고 재밌습니다
사람마다 코드 적는 말투(?)도 달라서 저의 코드 읽는 실력도 늘어가는 것 같아요

**Comment by tolluset** [comment]
[Went Well]
기한이 정해져있는거, 다른 분들의 풀이를 볼 수 있어서 좋았어요.

기한내에 풀어야하다보니 모르는 문제가 나오면 어느정도 검색 혹은 다른 분들의 풀이로 도움을 받았습니다. 이전에는 무기한으로 풀다보니 스스로 못 풀면 문제를 끝내지 않아서 실력이 늘지가 않더라구요. 

이제는 자주 나오는 문제 패턴은 살짝 보이기 시작했습니다 🫣

그리고 다른 분들의 풀이도 보면서 확실히 눈이 뜨이는 것 같습니다 ☺️

**Comment by DaleSeo** [comment]
[To improve] 5주차가 되니 코드 리뷰 참여도가 급격히 떨어지는 것 같아요. 코치님들의 활동도 눈이 띄게 줄었고, 멤버 분들의 상호 리뷰도 마찬가지입니다. 여기서 참여도가 더 떨어진다면 1기 때처럼 스터디의 전반적인 흐름에 병목 현상이 발생하지 않을까봐 걱정이네요. 저는 앞으로도 성심을 다해서 여러분의 코드를 리뷰를 해드릴 의향이 있지만, 우리 지속 가능하고 건강한 코드 리뷰 문화를 만들어 가고 있는지 다 같이 생각해보면 좋을 것 같습니다.

**Reply by TonyKim9401** [reply]
무료로 이런 좋은 스터디 진행하시는 달래님의 마음도 어느정도 이해는 됩니다.
2기 지원하면서 느낀건 지원자는 많고 심지어 참여하지 못한 인원도 계신걸로 아는데, 좀 더 지원자의 책임을 높이기 위해 1만원 정도의 적당한 회비를 걷으시고, 각 부문별로 `최고의 리뷰어`, `최다 문제 풀이`, `최다 문제 발표` 등을 만들어 회비를 모두 환원하는 식으로 가면 좀 더 참여도를 높일 수 있지 않을까 싶습니다. 개인의 이익을 지향하는것이 아닌 일종의 보증금? 같은 형식이고 참여도,기여도에 따라 모든 금액이 환원된다면 문제 없을것 같은 생각도 들구요!

**Reply by obzva** [reply]
개인적인 생각으로는 리뷰를 통한 지식 공유 및 학습이 문제 풀이 제출보다 더 중요한 것 같은데, 스터디가 점점 단순 숙제 검사처럼 되어가는 것 같아서 아쉽습니다 ㅎㅎㅎ

현생이 바빠 어쩔 수 없으시겠지만요 ㅜㅜ..

> 2기 지원하면서 느낀건 지원자는 많고 심지어 참여하지 못한 인원도 계신걸로 아는데

저도 2기 대기 명단에 이름을 올려놨다가 참여하게 되었습니다

참여하지 못한 인원들도 있었다는 사실을 기억해주세요~

**Comment by leokim0922** [comment]
[To improve] 초반을 제외하고 개인적인 일이 많이 일어나서 적극적인 참여를 많이 못해서 아쉬웠습니다. 여전히 현재진행형이지만 꾸준히 조금씩이라도 도와드리기 위해 노력하겠습니다 😄

**Comment by leokim0922** [comment]
[To improve] 개인적인 의견입니다만, 많은분들이 5주차쯤 오시면서 리트코드 문제들이 패턴을 가지고 있다는걸 느끼셨을거라 생각합니다. 이미 달레님의 알고달레가 충분히 좋은 정보를 제공하고있지만, 스터디 참여자분들이 원하신다면 Google Doc 같은 공유페이지를 하나 만들어서 지금까지 풀었던 문제들을 Categorisation 해보는거도 복기 + 차기수 가이드라인제공 측면에서 도움이 되지 않을까 생각합니다.

학교를 다니면서 LeetCode의 저 패턴을 스스로 잘 찾아내는 친구들도 많은반면, 상당히 늦게 혹은 계속 헤매는 친구들이 있는걸 봤는데요. 후자의 많은 사람들이 Design Gurus라는 플랫폼에서 패턴을 따로 돈내고 공부하는걸 봤습니다. 해당 플랫폼이 돈받고 하는걸 스터디 참여자들끼리 작성해서 소장하는거도 좋아보여서 제안드립니다 😃
